internal/abi/prim/prim.go

```diff
@@ -25,10 +25,18 @@ type U32 uint32
 // U64 is an unsigned 64 bit integer.
 type U64 uint64
 
+// Pointer is the type of a pointer
+type Pointer uint32
```

Comment:
One thing we could do is define this as

```go
type Pointer[_ any] uint32
```

and then instead of putting the old type in a comment it could be in the definition.

```go
func fastlyCacheGetHits(h cacheHandle, d /* *prim.U64 */ prim.Pointer) FastlyStatus
```

becomes

```go
func fastlyCacheGetHits(h cacheHandle, d prim.Pointer[*prim.U64]) FastlyStatus
```

Obviously it has no effect, but it maintains the annotation.

Reply:
Ooh, pretty.